x-pack/test/functional/services/ml/settings_calendar.ts

```diff
@@ -9,16 +9,73 @@ import expect from '@kbn/expect';
 
 import { FtrProviderContext } from '../../ftr_provider_context';
 import { MlCommonUI } from './common_ui';
+import { Navigation } from './navigation';
+import { MlADJobTable } from './job_table';
 
 export function MachineLearningSettingsCalendarProvider(
   { getService }: FtrProviderContext,
-  mlCommonUI: MlCommonUI
+  mlCommonUI: MlCommonUI,
+  navigation: Navigation,
+  jobTable: MlADJobTable
 ) {
   const testSubjects = getService('testSubjects');
   const retry = getService('retry');
   const comboBox = getService('comboBox');
 
   return {
+    async checkForCalendar(jobNumbers: number[], checkForExists: boolean = true) {
+      for await (const jobNumber of jobNumbers)
+        await jobTable.withDetailsOpen(`test_calendar_ad_${jobNumber}`, async () => {
+          await testSubjects[checkForExists ? 'existOrFail' : 'missingOrFail'](
+            jobTable.detailsSelector(
+              `test_calendar_ad_${jobNumber}`,
+              'mlJobRowDetailsSection-calendars'
+            )
+          );
+        });
+    },
+
+    async assertConnectedJobs(calendarId: string, regexp: RegExp) {
```

Comment:
A few comments that mostly repeat what I wrote on my first review:
1. I think this method is covering too much. The assertion for the calendar list is already a helper method and the assertion in the job list as well (see comment above for moving it to a different service). The navigation also already exists as well as assigning the calendar to a job group. The specific combination of these service methods is only used in one test file so should really live there.
2. Related to that: it's confusing that a method `assertXyz` actually modified the calendar and also leaves the test on a completely different page.
3. I think from a test file, instead of `ml.settingsCalendar.assertConnectedJobs` we want to call something like
    ```
    ml.settingsCalendar.assertConnectedJobs
    ml.navigation.navigateToAnomalyDetection
    ml.jobTable.assertJobRowCalendars
    ml.navigation.navigateToSettings
    ml.settingsCalendar.navigateToCalendarManagement
    ml.settingsCalendar.openCalendarEditForm
    ml.settingsCalendar.selectJobGroup
    ml.settingsCalendar.saveCalendar
    ml.settingsCalendar.assertConnectedJobs
    ml.navigation.navigateToAnomalyDetection
    ml.jobTable.assertJobRowCalendars
    ```

Reply:
Does [this](https://github.com/elastic/kibana/pull/178553/commits/b4529e5b647c5c15fa43cc68a11dc3095d748ad7) suffice?